### PR TITLE
Display details for selected node

### PR DESCRIPTION
**Pull request by florentc**
This is WIP to implement #86

### CODE REVIEW

**Comment by GuillaumeDesforges** [comment]
Thanks! Could you add screenshots/screen recording of before and after the PR changes please? 🙏

**Comment by florentc** [comment]
There is nothing of interest yet. I have implemented reacting to node selection but bringing in actual additional info is a bit more tricky. The representation of the graph in the front only contains package names, not their outputPath. So for now it only displays the name of the selected node as proof of concept. This is why this PR is only a draft, not ready for review.

Before
20230519_17h02m24s_grim

After
20230519_17h01m45s_grim

**Comment by dorranh** [comment]
Nice! One thought would be that if CystoscapeJs supports some form of vertex properties, we could potentially add UI support for those more generically first (even if that is just the name at the moment). Retrieving additional properties could potentially involve changes to the API (namely to ensure that it can return the property data in `QueryResult.cyto`) which is a mostly independent change that I think could be done separately from these updates.
 
https://github.com/tweag/nixpkgs-graph-explorer/blob/main/api/explorer/api/queries/query.py#L37

**Comment by florentc** [comment]
I agree. At the end of last week I tried to see how to bring the additional info (for now just the output path) in a nice way. In the end I figured that the most straightforward way would be to have an endpoint to query details of a specific package in the API (e.g. `/package/[package_name]`).

For now the graph (whose nodes are just package names without additional info) is built using a query that is not trivial to tweak in `getGraph` in `api.ts`. So the possibilities are: 

- storing all the info in the nodes of the graph but that would imply
- querying the info after the node is focused
   - internally within the front (we have the info after fetching all packages with the `/packages` API endpoint but we need to keep it available in the right component and perform the lookup)
   - use a new endpoint of the API such as described above (`/package/package_name`)

I think the most lightweight and maintainable solution is the last.

If we agree on this, then this PR could indeed just become about selecting a node in the graph and, for now, just display the name.

It works well in the current state using handlers passed along as parameters but @jlesquembre suggested to try using a custom event approach that is more idiomatic with web components. This would bring more consistency in event handling in the front.  I will try to implement this.

As for the visual/UI aspects, since we don't know yet how much (and what kind) of details will be displayed eventually, after the API provides more, I think I will stick to a simple bottom pane for now.

**Comment by GuillaumeDesforges** [comment]
Thanks @florentc 👍 I think this PR makes more sense if we at least display the "output path" of the selected node.

If this requires to edit the gremlin query, I'm sure @dorranh or @zz1874 can be of help 👌

**Comment by florentc** [comment]
Now multiselection is handled as well and the results are shown in a sidebar.

20230522_18h46m28s_grim

CSS wise, there remains an overflow bug to fix when too many results are selected at the same time.

Feature-wise, @GuillaumeDesforges, what do you think about having a new endpoint in the API to query details of a specific package as I described above? I feel like this would keep things simple in comparison to putting all the details in the graph and modifying the query that builds it. As for info beyond outputPath, the API will have to implement it beforehand because it is the only detail it gives about packages for now.

**Comment by GuillaumeDesforges** [comment]
Sure, i don't mind the implementation details, what matters is that we get the details displayed, including most urgently the output path

**Comment by florentc** [comment]
Rendering issues are fixed.

File: web/src/graph-viewer.ts [hunk 1]
```diff
@@ -170,6 +244,29 @@ export class GraphViewer extends LitElement {
   }
 }
 
+@customElement("packages-outline")
+export class PackagesOutline extends LitElement {
+  // TODO Eventually, this should be a collection of Pkg and contain actual details.
+  @property({ attribute: false }) packages: Array<string> = ["hello"];
```
Comment (GuillaumeDesforges) [inline web/src/graph-viewer.ts:250]: Shouldn't it be empty?

```suggestion
  @property({ attribute: false }) packages: Array<string> = [];
```

**Review by GuillaumeDesforges — CHANGES_REQUESTED** [review]

**Review by GuillaumeDesforges — APPROVED** [review]

**Comment by GuillaumeDesforges** [comment]
Thanks @florentc !

**Pull request merged**